scripts/build_test.py

```diff
@@ -14,29 +14,270 @@
+        self.assertEqual(calledProcessException.returncode, 1)
+
+    def test_generate_copy_tasks_for_fonts(self):
+        """Test _generate_copy_tasks_for_fonts to ensure that the number of copy
+        tasks matches the number of font files.
+        """
+        copy_tasks = collections.deque()
+        # Get all filepaths from manifest.json.
+        dependency_filepaths = build.get_dependencies_filepaths()
+        # Setup a sandbox folder for copying fonts.
+        target_fonts_dir = os.path.join('target', 'fonts', '')
+
+        self.assertEqual(len(copy_tasks), 0)
+        copy_tasks += build._generate_copy_tasks_for_fonts(
+            dependency_filepaths['fonts'], target_fonts_dir)
+        # Asserting the same number of copy tasks and number of font files.
+        self.assertEqual(len(copy_tasks), len(dependency_filepaths['fonts']))
+
     def test_insert_hash(self):
-        # pylint: disable=protected-access
-        self.assertEquals(build._insert_hash('file.js', '123456'),
-                          'file.123456.js')
-        self.assertEquals(build._insert_hash('path/to/file.js', '654321'),
-                          'path/to/file.654321.js')
-        self.assertEquals(build._insert_hash('file.min.js', 'abcdef'),
-                          'file.min.abcdef.js')
-        self.assertEquals(build._insert_hash('path/to/file.min.js', 'fedcba'),
-                          'path/to/file.min.fedcba.js')
-        # pylint: enable=protected-access
+        """Test _insert_hash to return correct filenames with provided hashes.
+        """
+        self.assertEqual(
+            build._insert_hash('file.js', '123456'), 'file.123456.js')
+        self.assertEqual(
+            build._insert_hash(
+                'path/to/file.js', '654321'), 'path/to/file.654321.js')
+        self.assertEqual(
+            build._insert_hash('file.min.js', 'abcdef'), 'file.min.abcdef.js')
+        self.assertEqual(
+            build._insert_hash(
+                'path/to/file.min.js', 'fedcba'), 'path/to/file.min.fedcba.js')
+
+    def test_ensure_files_exist(self):
+        """Test _ensure_files_exist raises exception with a non-existent
+        filepath.
+        """
+        random_filepaths = [
+            os.path.join(build.THIRD_PARTY_GENERATED_DEV_DIR, 'random1.js')]
+        with self.assertRaises(OSError) as fileNotExist:
+            build._ensure_files_exist(random_filepaths)
+        # Exception will be raised at first file determined to be non-existent.
+        self.assertTrue(
+            ('File %s does not exist.') % random_filepaths[0] in
+            fileNotExist.exception)
+
+    def test_get_file_count(self):
+        """Test get_file_count to return the correct number of files, excluding
+        file with extensions in FILE_EXTENSIONS_TO_IGNORE.
+        """
+        all_inclusive_file_count = 0
+        for _, _, files in os.walk(
+                build.EXTENSIONS_DIRNAMES_TO_DIRPATHS['dev_dir']):
```

Comment:
This and the following function (and maybe others) are a bit icky.

The tests should use a special test directory that's far away from any constants defined in build.py. Maybe you can put some stuff in core/tests and refer to that instead. (Feel free to make a build/ directory in there and then put various dir configurations, then do simple equality checks that count the files there or compare file counts directly.)

Reply:
I have setup a test dir inside core/tests/build and added the folder to .gitignore. All of the test functions now operate within that `core/test/build`